wallet-new/src/Cardano/Wallet/WalletLayer/Kernel/Wallets.hs

```diff
@@ -159,6 +163,22 @@ getWallets db =
     let allRoots = readAllHdRoots (hdWallets db)
     in IxSet.fromList . map (toV1Wallet db) . IxSet.toList $ allRoots
 
+
+-- | Gets Utxos per account of a wallet.
+getWalletUtxos :: Kernel.DB
+               -> V1.WalletId
+               -> Either GetUtxosError [(V1.Account, Utxo)]
+getWalletUtxos db (V1.WalletId wId) =
+    case decodeTextAddress wId of
+        Left _ -> Left (GetWalletUtxosWalletIdDecodingFailed wId)
+        Right rootAddr -> do
+            case getAccounts db (V1.WalletId wId) of
+                Left accountsError -> Left (GetUtxosErrorFromGetAccountsError accountsError)
+                Right accountsIxSet ->
+                    let hdRootId = HD.HdRootId . InDb $ rootAddr
+                        hdAccountId accountIndex = HD.HdAccountId hdRootId (HD.HdAccountIx accountIndex)
+                    in Right ( map (\acc -> (acc, accountUtxo db (hdAccountId $ V1.accIndex acc) ) ) $ IxSet.toList accountsIxSet )
```

Comment:
This uses the style of the wallet layer implementation as it was before I refactored it. Take a look at what it is now, the use of ExceptT can make a bit easier to write, and then you can also use from existing combinators such as `fromRootId`.